blogs/rtt/index.md

```diff
@@ -0,0 +1,157 @@
+
+For more detailed  explanation of smoothed RTT estimation,  refer to [Karn's algorithm paper](http://ccr.sigcomm.org/archive/1995/jan95/ccr-9501-partridge87.pdf)
+
+![TCP based RTT calculations](./images/tcp_rtt_calculations.png)
+
+### Why using `fentry` eBPF hook
+
+EBPF `fentry` programs have lower overhead as they trigger
+the hook before calling the kernel function of interest.
+
+In our implementation:
+
+1. Register and link `fentry` hook for kernel's `tcp_rcv_established()`
+
+```cgo
+SEC("fentry/tcp_rcv_established")
+int BPF_PROG(tcp_rcv_fentry, struct sock *sk, struct sk_buff *skb) {
+    if (sk == NULL || skb == NULL) {
+        return 0;
+    }
+    return calculate_flow_rtt_tcp(sk, skb);
+}
+```
+
+2. Reconstruct the Netobserv flow key, including incoming interface Layer2, Layer3, and Layer4 info.
+
+3. Match existing flows in the PerCPU hashmap flow table and enrich them with srtt info from TCP sockets. If
+   multiple SRTT values exist for the same flow, we take the maximum value.
+
+Currently, our approach calculates RTT only for the TCP packets so flows which are non TCP will not show RTT information.
+
+## Potential Use Cases
+
+Flow RTT capture from eBPF `flow_monitor` hookpoint can serve various purposes:
+
+- Network Monitoring: Gain insights into TCP handshakes, helping
+  network administrators identify unusual patterns, potential bottlenecks, or
+  performance issues.
+
+- Troubleshooting: Debug TCP-related issues by tracking latency and identifying
+  misconfigurations.
+
+## How to Enable RTT
+
+To enable this feature we need to create a flow collector object with the following
+fields enabled in eBPF config section as below:
+
+```yaml
+apiVersion: flows.netobserv.io/v1beta1
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  agent:
+    type: EBPF
+    ebpf:
+      features:
+        - FlowRTT
+```
+
+## A Quick Tour in the UI
+
+Once the `FlowRTT` feature is enabled, the OCP console plugin automatically adapts
+to provide additional filter and show information across Netflow Traffic page views.
+
+Open your OCP Console and move to
+`Administrator view` -> `Observe` -> `Network Traffic` page as usual.
+
+A new filter, `Flow RTT` is available in the common section:
+
+![flow rtt filter](./images/flow_rtt_filter.png)
+
+It will allow you capture any flow having a RTT more than a specific time in nanoseconds.
+
+For production users, filtering on the `TCP` protocol, `Ingress` direction,
+and looking for `FlowRTT` values greater than 10,000,000 nanoseconds (10ms)
+can help identify TCP flows with high latency.
+This filtering approach allows users to focus on specific network flows that may
+be experiencing significant delays.
+By setting a threshold of 10ms, you can efficiently isolate and address potential
+latency issues in your TCP traffic.
+
+### Overview
+
+New graphs are introduced in the `advanced options` -> `manage panels` popup:
```

Comment:
```suggestion
New graphs are introduced in the `Advanced options` -> `Manage panels` popup:
```

Reply:
In the UI, I think these are capitalized. And also, in product docs, we usually [bold UI elements](https://redhat-documentation.github.io/supplementary-style-guide/#user-interface-elements), and backticks are reserved for specifications and API objects. This isn't product docs but I just wanted to note that here in case you want to apply the same stylistic consistency.